### PR TITLE
Verify signatures from the ledger via provide public key.

### PR DESCRIPTION
**Pull request by jonored**
This updates the test script to remove all hardcoded public keys for the
test suite, and instead verifies ledger signatures with the public key
provided by the provide public key APDU, and also allows providing
non-default witnesses to allow checking the dao functions.

### CODE REVIEW

File: tests/lib.sh [hunk 1]
```diff
@@ -26,8 +26,11 @@ blake2b_lock_hash () {
 }
 
 check_signature () {
-  r_val=$(head -c64 <<<"$2")
-  s_val=$(tail -c+65 <<<"$2" | head -c64)
+  sig=$(tr -d -c '[:xdigit:]' <<<"$3")
+  echo "$sig" >> sigval
```
Comment (matthewbauer) [inline tests/lib.sh]: I think we may want to remove these temp files. It’s good for debugging but ends up littering your directory if you run the test suite outside of a nix builder.

File: tests/lib.sh [hunk 2]
```diff
@@ -36,12 +39,23 @@ check_signature () {
...
+  SIG="30 $(printf "%x" $(($rlen+$slen+4))) $rfmt $sfmt"
+  echo $SIG > sigdump
 
-  xxd -r -ps <<<"$1" | blake2b_lock_hash | xxd -p -r | openssl pkeyutl -verify -pubin -inkey tests/public_key_0_0.pem -sigfile <(xxd -r -ps <<<"$SIG")
+  xxd -r -ps <<<"$2" | tee dumpfile0 | blake2b_p | tee hashfile | xxd -p -r | openssl pkeyutl -verify -pubin -inkey <(get_key_in_pem $1) -sigfile <(xxd -r -ps <<<"$SIG")
+}
+
+get_key_in_pem() {
+  derivation=${1:-8002000015058000002c80000135800000000000000000000000}
+  result="$(apdu_with_clicks $derivation "rR" |& egrep "b'41.*'9000" | sed "s/^<= b'41//;s/'9000$//")"
```
Comment (matthewbauer) [inline tests/lib.sh]: Nice! Will test this on my ledger

File: tests/lib.sh (hunk 1 above)
Comment (jonored) [reply tests/lib.sh]: Whoops, yes.

File: tests/lib.sh (hunk 2 above)
Comment (jonored) [reply tests/lib.sh]: > public_key_der_prefix.der looks like it was added by mistake.

public_key_der_prefix.der is the prefix to convert from the raw key provided by the ledger to the ASN.1 format that openssl uses; we definitely need it. Could rename it.

**Pull request merged**